doc/array_and_matrix_manipulation.md

```diff
@@ -0,0 +1,271 @@
+% Array and Matrix Manipulation
+
+ArrayFire provides several different methods for manipulating arrays and matrices.
+The functionality includes:
+
+* [moddims()](#moddims) - change the dimensions of an array without changing the data
+* [flat()](#flat) - flatten an array to one dimension
+* [flip()](#flip) - flip an array along a dimension
+* [join()](#join) - join up to 4 arrays
+* [reorder()](#reorder) - changes the dimension order within the array
+* [shift()](#shift) - shifts data along a dimension
+* [tile()](#tile) - repeats an array along a dimension
+* [transpose()](#transpose) - performs a matrix transpose
+
+Below we provide several examples of these functions and their use.
+
+### moddims()
```

Comment:
That is true, but having these collected set of examples in one location is probably a good place to start for beginners and they can dwell into parameter documentation and additional examples in the hyperlinked function documentation page - at least that is how I picture it in the long run. I do agree that more examples in each function doc page are to be added yet.